packages/hardhat-core/src/internal/hardhat-network/provider/utils/makeFakeSignature.ts

```diff
@@ -18,8 +16,24 @@ export function makeFakeSignature(
   r: number;
   s: number;
 } {
+  const hashInputString = [
+    sender,
+    tx.nonce,
+    tx.gasLimit,
+    tx.value,
+    tx.to,
+    tx.data,
+    "gasPrice" in tx ? tx.gasPrice : "",
+    "chainId" in tx ? tx.chainId : "",
+    "maxPriorityFeePerGas" in tx ? tx.maxPriorityFeePerGas : "",
+    "maxFeePerGas" in tx ? tx.maxFeePerGas : "",
+    // note: accessList omitted because it doesn't serialize readily.
```

Comment:
Add access list.

Reply:
And use separate separators for each of the nested pieces of the access list, so we can differentiate them in the serialization. And add an explicit separator to that existing `join` so that it's clear to the reader.

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/3175/commits/39edd2d93bb34742b0db197326c64bb2f3eebd24